src/sqlast/query.rs

```diff
@@ -215,10 +227,7 @@ pub enum TableFactor {
         subquery: Box<SQLQuery>,
         alias: Option<TableAlias>,
     },
-    NestedJoin {
-        base: Box<TableFactor>,
-        joins: Vec<Join>,
-    },
+    NestedJoin(Box<TableReference>),
```

Comment:
Given https://github.com/andygrove/sqlparser-rs/pull/108#pullrequestreview-247376543 maybe call it `TableFactor::Nested`?

Reply:
Something about this whole design wasn't sitting right with me, so I decided to address the underlying issue you brought up in #108, rather than renaming this variant. I've pushed two commits that make it so that NestedJoin always contains a table reference with at least one join. A nice side effect of this work was supporting derived tables with set operations, like `SELECT * FROM (((SELECT 1) UNION (SELECT 2)) t1 NATURAL JOIN t2)`.